### PR TITLE
Binary search: show matching string

### PR DESCRIPTION
**Pull request by jstucke**

### CODE REVIEW

**Comment by dorpvom** [comment]
binary_search_screenshot

Can we have it so, that these "overfull hboxes" don't exist?

File: src/web_interface/components/database_routes.py [hunk 1]
```diff
@@ -98,31 +118,47 @@ def browse_searches(self):
...
         )
-        search_parameters['query'] = apply_filters_to_query(request, query)
-        if 'query_title' not in search_parameters:
-            search_parameters['query_title'] = search_parameters['query']
-        if request.args.get('date'):
-            search_parameters['query'] = self._add_date_to_query(search_parameters['query'], request.args.get('date'))
-        return search_parameters
+
+    @staticmethod
+    def _query_has_only_one_result(result_list, query):
```
Comment (dorpvom) [inline src/web_interface/components/database_routes.py:160]: This function seems to be unused in all of the project

File: src/web_interface/rest/rest_binary_search.py [hunk 2]
```diff
@@ -74,4 +74,11 @@ def get(self, search_id=None):
         if result is None:
             return error_message('The result is not ready yet or it has already been fetched', self.URL)
 
-        return success_message({'binary_search_results': result}, self.URL)
+        # the "new" binary search result has the structure {<uid>: {<rule>: [<str_match_data>]}}
```
Comment (dorpvom) [inline src/web_interface/rest/rest_binary_search.py:77]: Acknowledged. Although we should strongly consider dropping this compatibility #restapiv2

File: src/web_interface/templates/base.html [hunk 3]
```diff
@@ -138,6 +151,8 @@
                             class="fas fa-search"></i> Basic Search</a>
                     <a class="dropdown-item" href="/database/advanced_search" style="font-size: 14px"><i
                             class="fas fa-search-plus"></i> Advanced Search</a>
+                    <a class="dropdown-item" href="/database/graphql" style="font-size: 14px"><i
+                            class="fas fa-search-plus"></i> GraphQL Search</a>
```
Comment (dorpvom) [inline src/web_interface/templates/base.html:156]: Not in this PR

File: src/web_interface/templates/base.html [hunk 4]
```diff
@@ -49,62 +49,75 @@
...
-            .bg-fact {
-                  background-color: #255e54 !important;
-            }
-            .navbar-light .navbar-brand, .navbar-light .navbar-nav > li > a {
-                color: #ddd;
-            }
-            .navbar-light .navbar-brand:hover, .navbar-light .navbar-nav > li > a:hover {
-                color: #fff;
-            }
+        a:link{
```
Comment (dorpvom) [inline src/web_interface/templates/base.html:52]: Beurk. Changed indentation

**Review by dorpvom — CHANGES_REQUESTED** [review]
Please consider annotated requests

File: src/web_interface/templates/base.html (hunk 4 above)
Comment (jstucke) [reply src/web_interface/templates/base.html:52]: more like "fixed indentation": It was eight spaces here for no apparent reason

File: src/web_interface/templates/base.html (hunk 3 above)
Comment (jstucke) [reply src/web_interface/templates/base.html:156]: Whoopsie... removed

File: src/web_interface/rest/rest_binary_search.py (hunk 2 above)
Comment (jstucke) [reply src/web_interface/rest/rest_binary_search.py:77]: We need to add a GraphQL search endpoint anyway, so maybe we can combine this (and maybe also remove the "Advanced Search" endpoint then)

File: src/helperFunctions/yara_binary_search.py [hunk 5]
```diff
@@ -50,24 +51,55 @@ def _get_file_paths_of_files_included_in_fw(self, fw_uid: str) -> list[str]:
         return [self.fs_organizer.generate_path_from_uid(uid) for uid in self.db.get_all_files_in_fw(fw_uid)]
 
     @staticmethod
-    def _parse_raw_result(raw_result: str) -> dict[str, list[str]]:
+    def _parse_raw_result(raw_result: str, match_limit: int = 10) -> dict[str, dict[str, list[dict]]]:
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/helperFunctions/yara_binary_search.py:54]: Introduced a config variable for that

**Review by dorpvom — APPROVED** [review]

**Pull request merged**